src/js/stores/MesosLogStore.js

```diff
@@ -1,30 +1,62 @@
-import {Store} from 'mesosphere-shared-reactjs';
+import BaseStore from './BaseStore';
 
 import ActionTypes from '../constants/ActionTypes';
 import AppDispatcher from '../events/AppDispatcher';
 import Config from '../config/Config';
-import EventTypes from '../constants/EventTypes';
-import GetSetMixin from '../mixins/GetSetMixin';
+import EventTypes from '../constants/EventTypes';;
 import Item from '../structs/Item';
 import LogBuffer from '../structs/LogBuffer';
 import MesosLogActions from '../events/MesosLogActions';
 
 const MAX_FILE_SIZE = 50000;
 
-const MesosLogStore = Store.createStore({
-  storeID: 'mesosLog',
-
-  mixins: [GetSetMixin],
-
-  addChangeListener: function (eventName, callback) {
+class MesosLogStore extends BaseStore {
+
+  constructor() {
+    super(...arguments);
+
+    this.dispatcherIndex = AppDispatcher.register((payload) => {
```

Comment:
I didn't notice that the instance is exported, so that makes it kind of safe, true.
But to be honest, the only technically save solution I can think about to register the dispatcher-function is:
Outside of the class, using an instance created by a singleton.
Not in the constructor.

But yeah, not in this PR maybe.